### DISCUSSION TITLE
Maven Build Fails with Non-resolvable Parent POM in Vue-Webpack-Sample Project

### DISCUSSION THREAD

**Discussion by aimanskie** [exoplatform/exo-community#34] [Q&A]
Hi 👋 ! I'm trying to set up a Vue.js portlet according to the documentation provided by eXo Platform .
- https://docs.exoplatform.org/guide/developer-guide/intermediate-integration/front-end-application.html#deploy-and-test-the-application

I'm new to Java and Maven, but I need to get this project up and running. I am facing issues during the 
```
mvn clean package
```
Here's the error message I'm encountering:
```
[ERROR] Non-resolvable parent POM for com.acme.samples:vue-webpack-sample:1.0: 
Could not find artifact org.exoplatform:maven-exo-parent-pom:pom:26-M03 in central
(https://repo.maven.apache.org/maven2) and 
'parent.relativePath' points at wrong local POM @ line 2, column 11 -> [Help 2]
```
i'm using Apache Maven 3.9.4 and java 17.0.7 2023-04-18 LTS

Has anyone faced this issue before? How can I resolve this non-resolvable parent POM issue?

**Comment by rdenarie** [comment]
Hello @aimanskie 

The artifact org.exoplatform:maven-exo-parent-pom:pom is not available in maven central repository.

You need to add the exoplatform public repository in your maven settings.xml.

For that, find you settings.xml file (in linux generally, it is in ~/.m2/settings.xml. You will find information about file location and structure here : https://www.baeldung.com/maven-settings-xml)

Then in the "repositories" section, add this repository : 


>        <repository>
>           <id>exoPublic</id>
>           <url>https://repository.exoplatform.org/content/groups/public</url>
>           <layout>default</layout>
>         </repository>
> 

Then the artifact should be resolved.

I will add this information in documentation

**Reply by aimanskie** [reply]
Thank you @rdenarie for your guidance.

I updated my 'settings.xml' with the `<profiles>` and `<activeProfiles>` tags, and assigned the profile ID as `default`. This resolved my issue, and I was able to successfully build the WAR and JAR files.

Here's my updated 'settings.xml' for anyone who may encounter the same issue:

```xml
<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.0.0 https://maven.apache.org/xsd/settings-1.0.0.xsd">
  <profiles>
    <profile>
      <id>default</id>
      <repositories>
        <repository>
          <id>exoPublic</id>
          <url>https://repository.exoplatform.org/content/groups/public</url>
          <layout>default</layout>
        </repository>
      </repositories>
    </profile>
  </profiles>

  <activeProfiles>
    <activeProfile>default</activeProfile>
  </activeProfiles>
</settings>
